libnixt/test/SerializeSupport.cpp

```diff
@@ -0,0 +1,47 @@
+#include <gtest/gtest.h>
+
+#include "nixt/SerializeSupport.h"
+
+#include <bit>
+
+namespace nixt::serialize {
+
+/// These asserts are generated on little-endian machine.
+
+TEST(SerializeSupport, StringTable) {
+  StringTable ST;
+  std::ostringstream OS;
+  if (std::endian::native == std::endian::little) {
+    EXPECT_EQ(write(OS, ST, "Hello"), 0);
+    EXPECT_EQ(write(OS, ST, "World"), 13);
+    EXPECT_EQ(write(OS, ST, "Hello"), 0);
+    EXPECT_EQ(write(OS, ST, "World"), 13);
+    const char Expected[] = "\x5\0\0\0\0\0\0\0"
+                            "Hello"
+                            "\x5\0\0\0\0\0\0\0"
+                            "World";
+    EXPECT_TRUE(std::memcmp(Expected, OS.str().c_str(), sizeof(Expected)) == 0);
+  }
+}
+
+TEST(SerializeSupport, writeMisc) {
+  std::ostringstream OS;
+  EXPECT_EQ(write(OS, 1), 0);
+  EXPECT_EQ(write(OS, 2), 4);
+  EXPECT_EQ(write(OS, 3), 8);
+  EXPECT_EQ(write(OS, 4), 12);
+  EXPECT_EQ(write(OS, 5.5), 16);
+  EXPECT_EQ(write(OS, 6.5), 24);
+
+  if (std::endian::native == std::endian::little) {
```

Comment:
I think this should be `static_assert`, because we do not have big-endian machines now to test all branches. Currently "if-statement" cannot guarantee the code works correct on big-endian targets.